File: Factory/Payment.js
class upiPayment {
    pay(amount){
        console.log(`Payment done using upi for ${amount}`);
    }
};

class cardPayment { //Products
    pay(amount){
        console.log(`Payment done using card for ${amount}`);
    }
};

class netBankingPayment {
    pay(amount){
        console.log(`Payment done using net banking for ${amount}`);
    }
};

class wrongPayment {
    pay(amount){
        throw new Error("wrong payment type");
    }
}

class paymentFactory { //Factory
    static paymentType(type){
        switch(type){
            case 'upi':
                return new upiPayment();
            case 'card':
                return new cardPayment();
            case 'netBanking':
                return new netBankingPayment();
            default:
                return new wrongPayment();
        }
    }
}

const paymentClient = paymentFactory.paymentType('netBanking'); //Client
paymentClient.pay(300);

File: Factory/Payment.java
interface PaymentType{
    public void pay(int amt);
}
class upiPayment implements PaymentType{
    @Override
    public void pay(int amt){
        System.out.println("payement of "+amt+" is done using upi");
    }
}
class netBankingPayment implements PaymentType{
    @Override
    public void pay(int amt){
        System.out.println("payement of "+amt+" is done using net banking");
    }
}
class cardPayment implements PaymentType{
    @Override
    public void pay(int amt){
        System.out.println("payement of "+amt+" is done using card");
    }
}
class wrongPayment implements PaymentType{
    @Override
    public void pay(int amt){
        throw new Error("wrong payment method");
    }
}
class Factory{
    public static PaymentType getPayment(String type){
        switch(type){
            case "upi":
                return new upiPayment();
            case "netbanking":
                return new netBankingPayment();
            case "card":
                return new cardPayment();
            default:
                return new wrongPayment();
        }
    }
}
class Payment{
    public static void main(String args[]){
        PaymentType client = Factory.getPayment("upi");
        client.pay(300);
        PaymentType client2 = Factory.getPayment("");
        client2.pay(300);
    }
}